src/main/java/org/opensearch/knn/index/KNNIndexShard.java

```diff
@@ -126,17 +127,21 @@ public void warmup() throws IOException {
                     // was L2. So, if Space Type is not present, just fall back to L2
                     String spaceTypeName = fieldInfo.attributes().getOrDefault(SPACE_TYPE, SpaceType.L2.getValue());
                     SpaceType spaceType = SpaceType.getSpace(spaceTypeName);
-                    String engineFileName = buildEngineFileName(reader.getSegmentInfo().info.name,
-                            knnEngine.getLatestBuildVersion(), fieldInfo.name, fileExtension);
-
-                    engineFiles.putAll(reader.getSegmentInfo().files().stream()
-                            .filter(fileName -> fileName.equals(engineFileName))
-                            .map(fileName -> shardPath.resolve(fileName).toString())
-                            .filter(Objects::nonNull)
-                            .collect(Collectors.toMap(fileName -> fileName, fileName -> spaceType)));
+
+                    engineFiles.putAll(getEnginePaths(reader.getSegmentInfo().files(),
+                            reader.getSegmentInfo().info.name, fieldInfo.name, fileExtension, shardPath, spaceType));
                 }
             }
         }
         return engineFiles;
     }
+
+    protected Map<String, SpaceType> getEnginePaths(Collection<String> files, String segmentName, String fieldName,
+                                                   String fileExtension, Path shardPath, SpaceType spaceType) {
+        return files.stream()
+                .filter(fileName -> fileName.startsWith(buildEngineFilePrefix(segmentName)))
```

Comment:
We don't have to compute buildEngineFilePrefix for every file. This can be called once outside streams and used inside filer.

Reply:
Makes sense. Updated.